src/AnimationManager.ts

```diff
@@ -128,35 +113,183 @@ export class AnimationManager {
 			this.finish(!!beforeParam);
 		}
 	}
-	finish(isTrusted) {
+
+	public finish(isTrusted): void {
 		this._animateParam = null;
 		this.itm.setInterrupt(false);
 		this.em.triggerFinish(isTrusted);
 	}
-	private animateLoop(param: AnimationParam, complete: () => void) {
+
+	public getUserControl(param: AnimationParam) {
+		const userWish = param.setTo();
+		userWish.destPos = this.axm.get(userWish.destPos);
+		userWish.duration = minMax(
+			userWish.duration,
+			this._options.minimumDuration,
+			this._options.maximumDuration);
+		return userWish;
+	}
+
+	public animateTo(destPos: Axis, duration: number, option?: ChangeEventOption): void {
+		const param: AnimationParam = this.createAnimationParam(destPos, duration, option);
+		const depaPos = { ...param.depaPos };
+		const retTrigger = this.em.triggerAnimationStart(param);
+
+		// to control
+		const userWish = this.getUserControl(param);
+
+		// You can't stop the 'animationStart' event when 'circular' is true.
+		if (!retTrigger && this.axm.every(
+			userWish.destPos,
+			(v, opt) => isCircularable(v, opt.range, opt.circular as boolean[]))) {
+			console.warn("You can't stop the 'animation' event when 'circular' is true.");
+		}
+
+		if (retTrigger && !equal(userWish.destPos, depaPos)) {
+			const inputEvent = option && option.event || null;
+			this.animateLoop({
+				depaPos,
+				destPos: userWish.destPos,
+				duration: userWish.duration,
+				delta: this.axm.getDelta(depaPos, userWish.destPos),
+				isTrusted: !!inputEvent,
+				inputEvent,
+				input: option && option.input || null,
+			}, () => this.animationEnd());
+		}
+	}
+
+	public easing(p): number {
+		return p > 1 ? 1 : this._options.easing(p);
+	}
+
+	public setTo(pos: Axis, duration: number = 0) {
+		const axes: string[] = Object.keys(pos);
+		this.stop(axes);
+		const orgPos: Axis = this.axm.get(axes);
+
+		if (equal(pos, orgPos)) {
+			return this;
+		}
+		this.itm.setInterrupt(true);
+		let movedPos = filter(pos, (v, k) => orgPos[k] !== v);
+		if (!Object.keys(movedPos).length) {
+			return this;
+		}
+
+		movedPos = this.axm.map(movedPos, (v, opt) => {
+			const { range, circular } = opt;
+
+			if (circular && (circular[0] || circular[1])) {
+				return v;
+			} else {
+				return getInsidePosition(v, range, circular as boolean[]);
+			}
+		});
+
+		if (equal(movedPos, orgPos)) {
+			return this;
+		}
+
+		if (duration > 0) {
+			this.animateTo(movedPos, duration);
+		} else {
+			this.em.triggerChange(movedPos);
+			this.finish(false);
+		}
+
+		return this;
```

Comment:
You have to consider more when you create interfaces like this.
In this method, it can trigger "change" / "finish" event, but sometimes it can't like when the `pos` equals `orgPos`
The problem is, the user can't predict such occasions easily, and the method doesn't have any feedback about such cases.
So, one solution for this is might be returning a boolean value instead of `this`, whether it did actually animated to the desired position or not. But you can choose your own :)

Reply:
Currently, axes methods such as setTo and setBy collectively return an instance of the module itself, but changing this in the future seems to be a good discussion.